empress_gui.py

```diff
@@ -160,69 +140,126 @@ def __init__(self, master):
         self.recon_dup_label = tk.Label(self.recon_nums_frame)
         self.recon_trans_label = tk.Label(self.recon_nums_frame)
         self.recon_loss_label = tk.Label(self.recon_nums_frame)
+        self.recon_info_displayed = False
+        App.recon_graph = None
+
+    def init_view_solution_space(self):
+        # "View solution space" dropdown
+        self.view_solution_space_var = tk.StringVar(self.input_frame)
+        self.view_solution_space_var.set("View solution space")
+        self.view_solution_space_options = ["Entire space", "Clusters"]
+        self.view_solution_space_dropdown = tk.OptionMenu(self.input_frame, self.view_solution_space_var, *self.view_solution_space_options, command=self.select_from_view_solution_space_dropdown)
+        self.view_solution_space_dropdown.configure(width=15)
+        self.view_solution_space_dropdown.configure(state=tk.DISABLED)
+        self.view_solution_space_dropdown.grid(row=4, column=0)
 
         self.num_cluster_input = tk.IntVar()
         self.num_cluster = None
-
-        self.recon_info_displayed = False
-        self.recon_input = empress.ReconInputWrapper()
-        App.recon_graph = None
         App.clusters_list = []
         App.medians = None
 
+    def init_view_reconciliations(self):
+        # "View reconciliations" dropdown
+        self.view_reconciliations_var = tk.StringVar(self.input_frame)
+        self.view_reconciliations_var.set("View reconciliations")
+        self.view_reconciliations_options = ["One MPR", "One per cluster"]
+        self.view_reconciliations_dropdown = tk.OptionMenu(self.input_frame, self.view_reconciliations_var, *self.view_reconciliations_options, command=self.select_from_view_reconciliations_dropdown)
+        self.view_reconciliations_dropdown.configure(width=15)
+        self.view_reconciliations_dropdown.configure(state=tk.DISABLED)
+        self.view_reconciliations_dropdown['menu'].entryconfigure("One per cluster", state = "disabled")
+        self.view_reconciliations_dropdown.grid(row=5, column=0)
+
+    def init_view_pvalue_histogram(self):
+        # "View p-value histogram" button
+        self.view_pvalue_histogram_btn = tk.Button(self.input_frame, text="View p-value histogram", command=self.open_window_pvalue_histogram, state=tk.DISABLED, width=18)
+        self.view_pvalue_histogram_btn.grid(row=6, column=0)
+
+    def init_windows(self):
         self.cost_space_window = None
         self.entire_space_window = None
         self.set_num_cluster_window = None
         self.one_MPR_window = None
         self.view_solution_space_window = None
         self.view_reconciliations_window = None
         self.view_pvalue_histogram_window = None
+    
+    def refresh_when_reload_host(self):
+        if self.first_time_loading_files == True:
+            self.recon_input.read_host(self.host_file_path)
+            self.host_tree_info.destroy()
+        else:
+            if self.need_to_reset == True:
```

Comment:
remove `== True`

Reply:
It seems that this code block
```python
self.recon_input.read_host(self.host_file_path)
self.host_tree_info.destroy()
self.parasite_tree_info.destroy()
self.mapping_info.destroy()
```
is shared by both the if and else cases. It might be nice to move it out of if/else. See example below:
```python
if x:
    do_a()
    do_b()
else:
    do_b()
```
should be changed to
```python
if x:
    do_a()
do_b()
```
Please do the same for other `refresh_when_reload_*`.

Reply:
Done.